driver-core/src/main/com/mongodb/internal/connection/DefaultClusterFactory.java

```diff
@@ -96,4 +131,28 @@ public Cluster createCluster(final ClusterSettings clusterSettings, final Server
             }
         }
     }
+
+    private boolean noClusterEventListeners(final ClusterSettings clusterSettings, final ServerSettings serverSettings) {
+        return clusterSettings.getClusterListeners().isEmpty()
+                && serverSettings.getServerListeners().isEmpty()
+                && serverSettings.getServerMonitorListeners().isEmpty();
+    }
+
+    private static ClusterListener getClusterListener(final ClusterSettings clusterSettings) {
```

Comment:
Small behavioral change here.  Using multicaster when there is one listener registered, because the muilticaster will log exceptions thrown by the listeners, which should happen in all cases.  Didn't seem worth it to add a non-multicasting exception-logging implementation.